packages/sdk/test/e2e/node/apiClient.spec.ts

```diff
@@ -65,4 +71,33 @@ describe("ApiClient Tests - Node", () => {
     assert.equal(serverUsername, username);
     assert.equal(serverPassword, password);
   });
+
+  it("can connect to existing API using api key provider with api key in header", async function () {
+    // Arrange
+    const keyName = "x-api-key";
+    const keyVaule = "mock-api-key-vaule";
+    const apiKeyProvider = new ApiKeyProvider(keyName, keyVaule, ApiKeyLocation.Header);
+    const apiClient = createApiClient(apiBaseUrl, apiKeyProvider);
+
+    // Act
+    const res = await apiClient.get("/foo");
+
+    // Assert
+    assert.equal(res.data.url, "/foo");
+    assert.equal(res.data.requestHeader![keyName], keyVaule);
+  });
+
+  it("can connect to existing API using api key provider with api key in query parameter", async function () {
+    // Arrange
+    const keyName = "x-api-key";
+    const keyVaule = "mock-api-key-vaule";
+    const apiKeyProvider = new ApiKeyProvider(keyName, keyVaule, ApiKeyLocation.QueryParams);
+    const apiClient = createApiClient(apiBaseUrl, apiKeyProvider);
+
+    // Act
+    const res = await apiClient.get("/foo");
+
+    // Assert
+    assert.equal(res.data.url, "/foo?x-api-key=mock-api-key-vaule");
+  });
```

Comment:
let's add a test case that some special character should be encoded

Reply:
For example?

Reply:
something like `x&api&key`

Reply:
Fixed.